xbmc/CompileInfo.h

```diff
@@ -9,6 +9,16 @@
 #pragma once
 
 #include <string>
+#include <vector>
+
+/**
```

Comment:
Can't this be defined elsewhere? Seems odd to have the a "repo" definition in the compileinfo header. Shouldn't this be part of the add-on namespace? Not sure if defining it on a new header on the add-ons folder is cleaner though. @phunkyfish / @alwinus?

Reply:
Good point. It should be in defined as part of the addon headers. Just not sure exactly where.

Reply:
Understand what you mean, it would be much better to have the respective things confined to their place.
@phunkyfish what you thing do process independent also for addon side?

The only good advantage here is the overview of what is set by the compiler.

I think it is safest to set it using a compile value. If by means of an external location (e.g. advancedsettings.xml) it would be easier to attack.

Reply:
From my side :+1: to have on addon side.

The whole addon system need also be cleaned up to reduce the amount of external places (outside addons folder) where access them.

Reply:
Maybe compile info just reads a vector of string pairs. The repo info can be defined in the repos class and not used elsewhere.

Reply:
This sounds like a good solution to make this more generic without mentioning repos or add-ons. Could a solution be created that just processes vectors of pairs and assigns them to keys? Then we could have a generic method that returns that vector provided the key name, in this case "repos"?

Reply:
This is more complicated due to the vector containing the repo infos needing to be accessible statically. So we'll actually go with a header just for the struct where it is name-spaced correctly.

We can look at this again once the cleanup of the addon code happens.

Reply:
i moved it to the minor refactoring/cleanup commit